File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/housecleaning/HouseCleaningServiceFactory.java
package com.neatorobotics.sdk.android.robotservices.housecleaning;

/**
 * Neato
 * Created by Marco on 04/01/2017.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */

public class HouseCleaningServiceFactory {
    public static HouseCleaningService get(String serviceVersion) {
        if("basic-1".equalsIgnoreCase(serviceVersion)) return new HouseCleaningBasic1Service();
        else if("basic-2".equalsIgnoreCase(serviceVersion)) return new HouseCleaningBasic2Service();
        else if("basic-3".equalsIgnoreCase(serviceVersion)) return new HouseCleaningBasic3Service();
        else if("basic-4".equalsIgnoreCase(serviceVersion)) return new HouseCleaningBasic4Service();
        else if("minimal-2".equalsIgnoreCase(serviceVersion)) return new HouseCleaningMinimal2Service();
        else return null;
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/housecleaning/HouseCleaningServiceFactory.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.robotservices.housecleaning

import com.neatorobotics.sdk.android.robotservices.RobotServices


object HouseCleaningServiceFactory {
    fun get(serviceVersion: String): HouseCleaningService? {
        return when {
            RobotServices.VERSION_BASIC_1.equals(serviceVersion, ignoreCase = true) -> HouseCleaningBasic1Service()
            RobotServices.VERSION_BASIC_3.equals(serviceVersion, ignoreCase = true) -> HouseCleaningBasic3Service()
            RobotServices.VERSION_MINIMAL_2.equals(serviceVersion, ignoreCase = true) -> HouseCleaningMinimal2Service()
            RobotServices.VERSION_BASIC_4.equals(serviceVersion, ignoreCase = true) -> HouseCleaningBasic4Service()
            RobotServices.VERSION_ADVANCED_4.equals(serviceVersion, ignoreCase = true) -> HouseCleaningAdvanced4Service()
            else -> null
        }
    }
}
